# Exploring the Titanic in R #

### The sinking of the Titanic is one of the most infamous shipwrecks in history. On April 15, 1912, during her maiden voyage, the widely considered “unsinkable” RMS Titanic sank after colliding with an iceberg. Unfortunately, there weren’t enough lifeboats for everyone onboard, resulting in the death of 1,502 out of 2,224 passengers and crew. ###

### As well as a tragic disaster, the Titanic is a classic film, but did you know the Titanic is also a readily available and popular dataset that's been available for a number of years? This is what we'll be exploring today! ###

<div id = 'titanic'>
    <div class = "inline-block">
        <figure><img src = 'titanic_picture.jpg' width = "40%" height = "20%"></figure>
    </div>
    </div>

### It's one of many datasets that are built-in to R, we can access this by loading the [Titanic Package](https://cran.r-project.org/web/packages/titanic/index.html), and this is what we'll be using today to perform some data analyses in R. ###

## What will be covered in the session? ##
This session will be a hands-on exploratory data analyses, including packages commonly used in data analyses:
* [dplyr](http://dplyr.tidyverse.org/) for data wrangling.
* [ggplot2](http://ggplot2.tidyverse.org/) for data visualization.

The aim is to work through all pre-processing and cleaning techniques which we might want to consider when working with any dataset.

The packages are already installed within this environment, but if you wanted to install these in R, you can do so using `install.packages("dplyr")`.

Before we can get started with analyses we need to load the relevant R libraries.

### Load R libraries. ###

In [3]:
library(dplyr)
library(ggplot2)

We'll then load in the data. The Titanic data consists of two datasets, `train` and `test`, but we'll only use the train data today.

### Load Titanic Train data. ###

In [4]:
train <- read.csv("https://raw.githubusercontent.com/paulhendricks/titanic/master/inst/data-raw/train.csv")

A good first step in understanding your data is to check the structure of the data, such as the data type of each attribute, as well as how many rows and columns (observations and variables in R language) we can do this using `str()`.

### Check the Structure of Titanic data. ### 

In [5]:
str(train)

'data.frame':	891 obs. of  12 variables:
 $ PassengerId: int  1 2 3 4 5 6 7 8 9 10 ...
 $ Survived   : int  0 1 1 1 0 0 0 0 1 1 ...
 $ Pclass     : int  3 1 3 1 3 3 1 3 3 2 ...
 $ Name       : chr  "Braund, Mr. Owen Harris" "Cumings, Mrs. John Bradley (Florence Briggs Thayer)" "Heikkinen, Miss. Laina" "Futrelle, Mrs. Jacques Heath (Lily May Peel)" ...
 $ Sex        : chr  "male" "female" "female" "female" ...
 $ Age        : num  22 38 26 35 35 NA 54 2 27 14 ...
 $ SibSp      : int  1 1 0 1 0 0 0 3 0 1 ...
 $ Parch      : int  0 0 0 0 0 0 0 1 2 0 ...
 $ Ticket     : chr  "A/5 21171" "PC 17599" "STON/O2. 3101282" "113803" ...
 $ Fare       : num  7.25 71.28 7.92 53.1 8.05 ...
 $ Cabin      : chr  "" "C85" "" "C123" ...
 $ Embarked   : chr  "S" "C" "S" "S" ...


### What are the variables in the dataset? ###

* **PassengerId**: Serial Number
* **Survived**: Contains binary Values of 0 & 1. Passenger did not survive — 0, Passenger Survived — 1.
* **Pclass**: Ticket Class | 1st Class, 2nd Class or 3rd Class Ticket
* **Name**: Name of the passenger
* **Sex**: Male or Female
* **Age**: Age in years — Integer
* **SibSp**: No. of Siblings / Spouses — brothers, sisters and/or husband/wife
* **Parch**: No. of parents/children — mother/father and/or daughter, son
* **Ticket**: Serial Number
* **Fare**: Passenger fare
* **Cabin**: Cabin Number
* **Embarked**: Port of Embarkment | C- Cherbourg, Q — Queenstown, S — Southampton

Now we know which fields are included in the dataset, we might want to have a look at the raw data before we begin our analyses.

We can do so using `head` to show the first 6 rows, and `tail` to show the last 6 rows.

In [6]:
head(train)

PassengerId Survived Pclass
1 1           0        3     
2 2           1        1     
3 3           1        3     
4 4           1        1     
5 5           0        3     
6 6           0        3     
  Name                                                Sex    Age SibSp Parch
1 Braund, Mr. Owen Harris                             male   22  1     0    
2 Cumings, Mrs. John Bradley (Florence Briggs Thayer) female 38  1     0    
3 Heikkinen, Miss. Laina                              female 26  0     0    
4 Futrelle, Mrs. Jacques Heath (Lily May Peel)        female 35  1     0    
5 Allen, Mr. William Henry                            male   35  0     0    
6 Moran, Mr. James                                    male   NA  0     0    
  Ticket           Fare    Cabin Embarked
1 A/5 21171         7.2500       S       
2 PC 17599         71.2833 C85   C       
3 STON/O2. 3101282  7.9250       S       
4 113803           53.1000 C123  S       
5 373450            8.0500       S       
6 330877            8.4583       Q

In [7]:
tail(train)

PassengerId Survived Pclass Name                                     Sex   
886 886         0        3      Rice, Mrs. William (Margaret Norton)     female
887 887         0        2      Montvila, Rev. Juozas                    male  
888 888         1        1      Graham, Miss. Margaret Edith             female
889 889         0        3      Johnston, Miss. Catherine Helen "Carrie" female
890 890         1        1      Behr, Mr. Karl Howell                    male  
891 891         0        3      Dooley, Mr. Patrick                      male  
    Age SibSp Parch Ticket     Fare   Cabin Embarked
886 39  0     5     382652     29.125       Q       
887 27  0     0     211536     13.000       S       
888 19  0     0     112053     30.000 B42   S       
889 NA  1     2     W./C. 6607 23.450       S       
890 26  0     0     111369     30.000 C148  C       
891 32  0     0     370376      7.750       Q

### Check for missing values ###

We know which fields we have in the dataset, and the size of the dataset, but it's important to consider any missing data.

We can do this using a combination of `is.na` and `colSums`.

In [ ]:
colSums(is.na(train))

We can see that `Age` has 177 missing values in our Train data. For other values we may want to impute missing values, but for missing ages, we'll leave them as blank.

### Extract Title from the Name field ###

We can extract title from `Name` using `gsub`. The gsub function replaces all instances of the pattern provided in the specified column of the dataframe. In this instance, we're looking for the characters `,` and `.` and replacing anything outside of this with blank space. 

* `(.*, )` -> any text before and including a comma and space
* `(\\..*)` -> any text after and including a dot
* The `|` -> an or operator
* The gsub will match the first OR second pattern

'Bloggs, Mr. Joe' would become 'Mr' and would be created as a column called `Title` in the Train dataset.

In [8]:
train$Title <- gsub('(.*, )|(\\..*)', '', train$Name)

We can use `head` again to glimpse at our data to check if this new column is as we'd expect.

In [9]:
head(train)

PassengerId Survived Pclass
1 1           0        3     
2 2           1        1     
3 3           1        3     
4 4           1        1     
5 5           0        3     
6 6           0        3     
  Name                                                Sex    Age SibSp Parch
1 Braund, Mr. Owen Harris                             male   22  1     0    
2 Cumings, Mrs. John Bradley (Florence Briggs Thayer) female 38  1     0    
3 Heikkinen, Miss. Laina                              female 26  0     0    
4 Futrelle, Mrs. Jacques Heath (Lily May Peel)        female 35  1     0    
5 Allen, Mr. William Henry                            male   35  0     0    
6 Moran, Mr. James                                    male   NA  0     0    
  Ticket           Fare    Cabin Embarked Title
1 A/5 21171         7.2500       S        Mr   
2 PC 17599         71.2833 C85   C        Mrs  
3 STON/O2. 3101282  7.9250       S        Miss 
4 113803           53.1000 C123  S        Mrs  
5 373450            8.0500       S        Mr   
6 330877            8.4583       Q        Mr

To get an overview of the data, we can use the function `summary`.

### Summarize Titanic data ###

In [10]:
summary(train)

  PassengerId       Survived          Pclass          Name          
 Min.   :  1.0   Min.   :0.0000   Min.   :1.000   Length:891        
 1st Qu.:223.5   1st Qu.:0.0000   1st Qu.:2.000   Class :character  
 Median :446.0   Median :0.0000   Median :3.000   Mode  :character  
 Mean   :446.0   Mean   :0.3838   Mean   :2.309                     
 3rd Qu.:668.5   3rd Qu.:1.0000   3rd Qu.:3.000                     
 Max.   :891.0   Max.   :1.0000   Max.   :3.000                     
                                                                    
     Sex                 Age            SibSp           Parch       
 Length:891         Min.   : 0.42   Min.   :0.000   Min.   :0.0000  
 Class :character   1st Qu.:20.12   1st Qu.:0.000   1st Qu.:0.0000  
 Mode  :character   Median :28.00   Median :0.000   Median :0.0000  
                    Mean   :29.70   Mean   :0.523   Mean   :0.3816  
                    3rd Qu.:38.00   3rd Qu.:1.000   3rd Qu.:0.0000  
                    Max.   :80.00 

Now do the same using a pipe `%>%`, one of the handiest tools in Tidyverse.

In [ ]:
train %>% 
  summary()

**Recap:**

* We've imported Titanic train data and viewed some of the data.
* We've used the pipe operator `%>%`.

**Next:**

* Data wrangling with `dplyr` to:
* **filter** our data to areas of interest.
* **arrange** the data to generate new insights.
* create new features by **mutating** existing ones.

## Wrangle data ##

We'll start pulling out some initial insights using `dplyr` functionality, such as `filter`, `arrange` and `mutate`.

These are really useful if we wanted to limit (filter) the data to only the observations of interest, or order by a particular value (arrange) or create new columns (mutate).

We can read `dplyr` code like a sentence: start with our "train" data (%>% and then) filter to only return those records where the sex is male `(filter(Sex == "male")`.

Filter to return all "male" rows

In [11]:
train %>% 
  filter(Sex == "male")

PassengerId Survived Pclass Name                                    Sex 
1    1          0        3      Braund, Mr. Owen Harris                 male
2    5          0        3      Allen, Mr. William Henry                male
3    6          0        3      Moran, Mr. James                        male
4    7          0        1      McCarthy, Mr. Timothy J                 male
5    8          0        3      Palsson, Master. Gosta Leonard          male
6   13          0        3      Saundercock, Mr. William Henry          male
7   14          0        3      Andersson, Mr. Anders Johan             male
8   17          0        3      Rice, Master. Eugene                    male
9   18          1        2      Williams, Mr. Charles Eugene            male
10  21          0        2      Fynney, Mr. Joseph J                    male
11  22          1        2      Beesley, Mr. Lawrence                   male
12  24          1        1      Sloper, Mr. William Thompson            male
13  27          0        3      Emir, Mr. Farred Chehab                 male
14  28          0        1      Fortune, Mr. Charles Alexander          male
15  30          0        3      Todoroff, Mr. Lalio                     male
16  31          0        1      Uruchurtu, Don. Manuel E                male
17  34          0        2      Wheadon, Mr. Edward H                   male
18  35          0        1      Meyer, Mr. Edgar Joseph                 male
19  36          0        1      Holverson, Mr. Alexander Oskar          male
20  37          1        3      Mamee, Mr. Hanna                        male
21  38          0        3      Cann, Mr. Ernest Charles                male
22  43          0        3      Kraeff, Mr. Theodor                     male
23  46          0        3      Rogers, Mr. William John                male
24  47          0        3      Lennon, Mr. Denis                       male
25  49          0        3      Samaan, Mr. Youssef                     male
26  51          0        3      Panula, Master. Juha Niilo              male
27  52          0        3      Nosworthy, Mr. Richard Cater            male
28  55          0        1      Ostby, Mr. Engelhart Cornelius          male
29  56          1        1      Woolner, Mr. Hugh                       male
30  58          0        3      Novel, Mr. Mansouer                     male
<U+22EE>   <U+22EE>    <U+22EE> <U+22EE> <U+22EE>                                <U+22EE>
548 841         0        3      Alhomaki, Mr. Ilmari Rudolf             male
549 842         0        2      Mudd, Mr. Thomas Charles                male
550 844         0        3      Lemberopolous, Mr. Peter L              male
551 845         0        3      Culumovic, Mr. Jeso                     male
552 846         0        3      Abbing, Mr. Anthony                     male
553 847         0        3      Sage, Mr. Douglas Bullen                male
554 848         0        3      Markoff, Mr. Marin                      male
555 849         0        2      Harper, Rev. John                       male
556 851         0        3      Andersson, Master. Sigvard Harald Elias male
557 852         0        3      Svensson, Mr. Johan                     male
558 858         1        1      Daly, Mr. Peter Denis                   male
559 860         0        3      Razi, Mr. Raihed                        male
560 861         0        3      Hansen, Mr. Claus Peter                 male
561 862         0        2      Giles, Mr. Frederick Edward             male
562 865         0        2      Gill, Mr. John William                  male
563 868         0        1      Roebling, Mr. Washington Augustus II    male
564 869         0        3      van Melkebeke, Mr. Philemon             male
565 870         1        3      Johnson, Master. Harold Theodor         male
566 871         0        3      Balkic, Mr. Cerin                       male
567 873         0        1      Carlsson, Mr. Frans Olof        

**Does our data give us any indication about whether men are likely to have survived or not?**

An initial view of the data seems to suggest that men typically did not survive (at least in the data we can see here).

**TO DO: Write the code to filter to filter to only return females**

**Does our data give us any indication about whether women are likely to have survived or not?**

An initial view of the data suggests that a lot of women did survive. We'll come on to this in more detail a little bit later.

What if we wanted to `arrange` our observations by increasing `Fare` to see if we can notice any trends? We can use `arrange` to achieve this.

Arrange by increasing Fare.

In [12]:
train %>%
  arrange(Fare)

PassengerId Survived Pclass
1   180         0        3     
2   264         0        1     
3   272         1        3     
4   278         0        2     
5   303         0        3     
6   414         0        2     
7   467         0        2     
8   482         0        2     
9   598         0        3     
10  634         0        1     
11  675         0        2     
12  733         0        2     
13  807         0        1     
14  816         0        1     
15  823         0        1     
16  379         0        3     
17  873         0        1     
18  327         0        3     
19  844         0        3     
20  819         0        3     
21  203         0        3     
22  372         0        3     
23  144         0        3     
24  655         0        3     
25  412         0        3     
26  826         0        3     
27  130         0        3     
28  805         1        3     
29  478         0        3     
30  132         0        3     
<U+22EE>   <U+22EE>    <U+22EE> <U+22EE>
862 196         1        1     
863 298         0        1     
864 306         1        1     
865 499         0        1     
866 709         1        1     
867 269         1        1     
868 333         0        1     
869 610         1        1     
870 319         1        1     
871 857         1        1     
872 690         1        1     
873 731         1        1     
874 780         1        1     
875 378         0        1     
876 528         0        1     
877 381         1        1     
878 558         0        1     
879 701         1        1     
880 717         1        1     
881 119         0        1     
882 300         1        1     
883 312         1        1     
884 743         1        1     
885  28         0        1     
886  89         1        1     
887 342         1        1     
888 439         0        1     
889 259         1        1     
890 680         1        1     
891 738         1        1     
    Name                                                  Sex    Age   SibSp
1   Leonard, Mr. Lionel                                   male   36.0  0    
2   Harrison, Mr. William                                 male   40.0  0    
3   Tornquist, Mr. William Henry                          male   25.0  0    
4   Parkes, Mr. Francis "Frank"                           male     NA  0    
5   Johnson, Mr. William Cahoone Jr                       male   19.0  0    
6   Cunningham, Mr. Alfred Fleming                        male     NA  0    
7   Campbell, Mr. William                                 male     NA  0    
8   Frost, Mr. Anthony Wood "Archie"                      male     NA  0    
9   Johnson, Mr. Alfred                                   male   49.0  0    
10  Parr, Mr. William Henry Marsh                         male     NA  0    
11  Watson, Mr. Ennis Hastings                            male     NA  0    
12  Knight, Mr. Robert J                                  male     NA  0    
13  Andrews, Mr. Thomas Jr                                male   39.0  0    
14  Fry, Mr. Richard                                      male     NA  0    
15  Reuchlin, Jonkheer. John George                       male   38.0  0    
16  Betros, Mr. Tannous                                   male   20.0  0    
17  Carlsson, Mr. Frans Olof                              male   33.0  0    
18  Nysveen, Mr. Johan Hansen                             male   61.0  0    
19  Lemberopolous, Mr. Peter L                            male   34.5  0    
20  Holm, Mr. John Fredrik Alexander                      male   43.0  0    
21  Johanson, Mr. Jakob Alfred                            male   34.0  0    
22  Wiklund, Mr. Jakob Alfred                             male   18.0  1    
23  Burke, Mr. Jeremiah                                   male   19.0  0    
24  Hegarty, Miss. Hanora "Nora"                          female 18.0  0    
25  Hart, Mr. Henry                                       male     NA  0    
26  Flynn

**Does our data give us any indication about whether fare paid has an effect on the chance of survival or not?**

From the data we can see, a lot of the people who paid less did not survive the disaster. That's something interesting that we've just discovered just by re-arranging our data.

The second half of the dataset (those who paid a higher fare) we can see a lot of the people who paid more did survive the disaster. We can confirm this using `arrange` in descending order.

In [13]:
train %>% 
  arrange(desc(Fare))

PassengerId Survived Pclass
1   259         1        1     
2   680         1        1     
3   738         1        1     
4    28         0        1     
5    89         1        1     
6   342         1        1     
7   439         0        1     
8   312         1        1     
9   743         1        1     
10  119         0        1     
11  300         1        1     
12  381         1        1     
13  558         0        1     
14  701         1        1     
15  717         1        1     
16  528         0        1     
17  378         0        1     
18  690         1        1     
19  731         1        1     
20  780         1        1     
21  319         1        1     
22  857         1        1     
23  269         1        1     
24  333         0        1     
25  610         1        1     
26  298         0        1     
27  306         1        1     
28  499         0        1     
29  709         1        1     
30   32         1        1     
<U+22EE>   <U+22EE>    <U+22EE> <U+22EE>
862 885         0        3     
863 478         0        3     
864 130         0        3     
865 805         1        3     
866 826         0        3     
867 412         0        3     
868 144         0        3     
869 655         0        3     
870 203         0        3     
871 372         0        3     
872 819         0        3     
873 844         0        3     
874 327         0        3     
875 873         0        1     
876 379         0        3     
877 180         0        3     
878 264         0        1     
879 272         1        3     
880 278         0        2     
881 303         0        3     
882 414         0        2     
883 467         0        2     
884 482         0        2     
885 598         0        3     
886 634         0        1     
887 675         0        2     
888 733         0        2     
889 807         0        1     
890 816         0        1     
891 823         0        1     
    Name                                                  Sex    Age   SibSp
1   Ward, Miss. Anna                                      female 35.00 0    
2   Cardeza, Mr. Thomas Drake Martinez                    male   36.00 0    
3   Lesurer, Mr. Gustave J                                male   35.00 0    
4   Fortune, Mr. Charles Alexander                        male   19.00 3    
5   Fortune, Miss. Mabel Helen                            female 23.00 3    
6   Fortune, Miss. Alice Elizabeth                        female 24.00 3    
7   Fortune, Mr. Mark                                     male   64.00 1    
8   Ryerson, Miss. Emily Borie                            female 18.00 2    
9   Ryerson, Miss. Susan Parker "Suzette"                 female 21.00 2    
10  Baxter, Mr. Quigg Edmond                              male   24.00 0    
11  Baxter, Mrs. James (Helene DeLaudeniere Chaput)       female 50.00 0    
12  Bidois, Miss. Rosalie                                 female 42.00 0    
13  Robbins, Mr. Victor                                   male      NA 0    
14  Astor, Mrs. John Jacob (Madeleine Talmadge Force)     female 18.00 1    
15  Endres, Miss. Caroline Louise                         female 38.00 0    
16  Farthing, Mr. John                                    male      NA 0    
17  Widener, Mr. Harry Elkins                             male   27.00 0    
18  Madill, Miss. Georgette Alexandra                     female 15.00 0    
19  Allen, Miss. Elisabeth Walton                         female 29.00 0    
20  Robert, Mrs. Edward Scott (Elisabeth Walton McMillan) female 43.00 0    
21  Wick, Miss. Mary Natalie                              female 31.00 0    
22  Wick, Mrs. George Dennick (Mary Hitchcock)            female 45.00 1    
23  Graham, Mrs. William Thompson (Edith Junkins)         female 58.00 0    
24  Graham, Mr. George Edward                             male   38.00 0    
25  Shutes, Miss. Elizabeth W                             female 40.00 0    
26  Allis

## Creating new variables using `mutate` ## 

We can also create new variables to add to our dataset, to give us new insights.

We know that the variable "Parch" is the number of parents and children while "SibSp" is the number of siblings and spouses. 

We can add these together to c a new variable "FamSize". To do so, we `mutate` the original variables into the new one.

Create new column `FamSize` (size of family)

In [ ]:
train %>% 
  mutate(FamSize = Parch + SibSp)

**Is it possible that larger families have a lower rate of survival?**

To test this, we can use our new variable "FamSize" and arrange by decreasing "FamSize".

In [ ]:
train %>% 
  mutate(FamSize = Parch + SibSp) %>% 
  arrange(desc(FamSize))

**Does our data give us any indication about whether larger families had an impact on the chance of survival or not?**

From the data we can see, some families had 7 or 10 members on-board, and all of the largest families we can see did not survive.

We can also use `mutate` to amend existing columns, we might want to re-assign 0 and 1 in the "Survived" column to a Yes or No.

Turn numerical values of Survived column to "No" & "Yes" (but assign to a new dataframe)

In [ ]:
survived <- train %>%
  mutate(Survived = ifelse(Survived == 0, "No", "Yes"))

In [ ]:
head(survived)

Notice that the Survived column has amended 0 to No, and 1 to Yes, which makes the dataset a little more user-friendly.

**Recap:**

* We've imported our Titanic data and viewed some of the data
* We've used the pipe operator `%>%`
* We can now use `dplyr` to wrangle our data to `filter`, `arrange` and create new features by `mutating` existing ones.

**Next:**

* We'll plot our data using `ggplot2` to get a visual view to derive new insights.

## Plotting data

To plot our data using `ggplot2`, we need to specify three things:

* Data
* Aesthetics (e.g. what's on the x-axis and y-axis)
* Plot (e.g. barplot, scatterplot)

**About ggplot:**

* We can read the `ggplot2` code like a sentence to understand what is happening: 

`ggplot(train, aes(x = Age, y = Fare)) +
  geom_point()`

**How to read the code**

We take the titanic `train` data and we map `Age` to the x axis and `Fare` to the y axis, we then add points on as the layer of the plot.

The first question that springs to mind regarding the Titanic is **"How many people survived?"**

Plot barplot of passenger surivival rate.

In [ ]:
ggplot(survived, aes(x = Survived)) +
  geom_bar()

This is useful, but it's quite tricky to see from this plot to know what proportion survived. We can make this clearer by returning the proportion. We can do so using the `prop.table` function.

In [ ]:
prop.table(table(survived$Survived))

We can see that only 38.4% of all passengers who on-boarded the Titanic survived.

We might want to investigate if the Age of any given passenger correlates with the fare paid. We can do this using a scatterplot, know as `geom_point` in `ggplot`.

Scatter plot of `Age` vs `Fare`.

In [ ]:
ggplot(survived, aes(x = Age, y = Fare)) +
  geom_point()

*We receive a warning above the plot, the 177 records have been removed with missing values. This is because of those NULL values we identified earlier.*

**Does our data give us any indication about whether there was any correlation between fare paid and age of the passenger?**

We can see that the majority of points within the data have a fare of £100 or less. There are also two notable outliers who had a fare greater than £500!

If we use the previous barplot and colour each point by `Sex` to see if there are any correlations between `Sex`, `Age` and `Fare`.

Scatter plot of `Age` vs `Fare` coloured by `Sex`.

In [ ]:
ggplot(survived, aes(x = Age, y = Fare, color = Sex)) +
  geom_point()

The plot above is useful, we can now see the data points are coloured by sex, as well as our original fare and age, but what if we wanted to add the Survived variable to the plot, to see if that shows us anything of interest?

**We can use `facet_grid` to produce multiple plots at the same time**

Scatter plot of Age vs Fare colored by Sex faceted by Survived.

In [ ]:
ggplot(survived, aes(x = Age, y = Fare, color = Sex)) +
  geom_point() +
  facet_grid(~Survived)

**Does our data give us any indication about whether Sex had an impact on the chance of survival or not?** 

We see that the survival rate amongst women was significantly higher when compared to men.

It is believed that in case of rescue operations during disasters, women's safety is prioritised. It certainly looks as though this is the case, with women much more likely to survive.

We can create a bar plot to see if this was the case overall, by plotting Sex and whether they survived or not.

Plot barplot of passenger `Sex` & fill according to `Survival`.

In [ ]:
ggplot(survived, aes(x = Sex, fill = Survived)) +
  geom_bar()

**Does our data give us any indication about whether Sex had an impact on the chance of survival or not?** 

We see that the survival rate amongst women was significantly higher when compared to men.

Can we see any differences depending on the type of ticket they held? This can be determined using the `pclass` variable e.g. 1st, 2nd or 3rd class.

**TO DO: Using the created survivded dataset, plot pclass against survived column using as a barplot.**

In [ ]:
#ggplot(survived, aes(x = Pclass, fill = Survived)) +
#  geom_bar()

**Does our data give us any indication about whether class had an impact on the chance of survival or not?** 

There were 3 groups of passengers, depending upon the class they were travelling in - 1st class, 2nd class and 3rd class. 

We see that over 50% of the passengers were travelling in the 3rd class, but a much higher proportion of passengers in 1st class survived.

Could it be preferential treatment why the passengers travelling first class, or could it be because the 3rd class compartments were in the lower extremity of the ship? Or could it be down to the placement of the lifeboats and how accessible they are to each class?

TO DO: *Add in picture of titanic classes*

In [ ]:
ggplot(survived, aes(x = Age, fill = Survived)) +
  geom_histogram(binwidth = 3, position = "fill") +
  ylab("Frequency")

**Does our data give us any indication about whether age had an impact on the chance of survival or not?** 

We can see that young children, particularly up to 10 years old, had a greater chance of survival. From the previous plot we see that women are more likely to survive than men. 

The captain explicitly issued an order for women and children to be saved first. As a result, the survival rate for women was three times higher than for men.

**Recap:**

* We've imported our Titanic data and viewed some of the data
* We've used the pipe operator `%>%`
* We can use `dplyr` to wrangle our data to `filter`, `arrange` and create new features by `mutating` existing ones.
* We've plotted our data using `ggplot2`.

**Next:**

* Learn how to `summarise` our data.


## Summarizing and grouping data ##

The summarize function reduces a data frame to a summary of just one vector or value. Many times, these summaries are calculated by grouping observations using a factor or categorical variables first. 

Use the `summarise` verb to find out the mean fare paid.

In [ ]:
train %>% 
  summarise(meanFare = mean(Fare))

We can use the `filter` and `summarise` functions to find out the mean fare paid among men.

In [ ]:
train %>% 
  filter(Sex == "male") %>% 
  summarise(meanFare = mean(Fare))

Use the `filter` and `summarise` verbs to find out the mean fare paid among women.

In [ ]:
#train %>% 
#  filter(Sex == "female") %>% 
#  summarise(meanFare = mean(Fare))

We can extend on this again, using the `filter` and `summarise` functions to find out the mean fare paid among women **and** how many women survived.

In [ ]:
train %>% 
  filter(Sex == "female") %>%
  summarise(meanFare = mean(Fare), 
            NumSurv = sum(Survived))

We can use Use the `group_by` and `summarise` functions to find the mean fare and number of survivors and group by `Sex`.

In [ ]:
train %>% 
  group_by(Sex) %>% 
  summarise(meanFare = mean(Fare),
            NumSurv = sum(Survived))

Use the `group_by` and `summarise` functions to find the mean fare and proportion of survivors and group by `Sex`.

In [ ]:
train %>% 
  group_by(Sex) %>% 
  summarise(meanFare = mean(Fare),
            NumSurv = sum(Survived)/n())

Can we do the same using PClass to see if the class of ticket has any significant impact on survival?

In [ ]:
train %>% 
  group_by(Pclass) %>% 
  summarise(meanFare = mean(Fare),
            NumSurv = sum(Survived)/n())

**Does our data give us any indication about whether class had an impact on the chance of survival or not?**

* Around 6 in 10 passengers within first class survived
* Around 5 in 10 passengers within second class survived
* Around 2 in 10 passengers within third class survived 

## In Summary ##

* We've imported the Titanic train data and viewed some of the data.
* We've used the pipe operator `%>%`.
* We can use `dplyr` to wrangle our data to `filter`, `arrange` and create new features by `mutating` existing ones.
* We can plot our data using `ggplot2`.
* We can summarise our data using `summarise`.

## Interested to know more? ##

Feel free to play around and do your own exploratory analysis, thinking about which fields you could explore or combine to create new insights.

## Thanks for joining! ##